Branch-SDK/src/io/branch/referral/BranchActivityLifecycleObserver.java

```diff
@@ -0,0 +1,148 @@
+package io.branch.referral;
+
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.app.Application;
+import android.content.Context;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+
+import java.lang.ref.WeakReference;
+
+import io.branch.indexing.ContentDiscoverer;
+
+/**
+ * <p>Class that observes activity life cycle events and determines when to start and stop
+ * session.</p>
+ */
+@TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH)
```

Comment:
Our SDK has a minimum of 16 now.   ICE_CREAM_SANDWICH is 14 so we don't need this check moving forward.

(In fact, there are still 8 of these that I wouldn't mind cleaning up while we are at it)